spec/unit/application/apply_spec.rb

```diff
@@ -171,6 +178,30 @@
       @apply.run_command
     end
 
+    it "should use a filesystem lock to restrict multiple processes running" do
+      @apply.expects(:lock)
+      @apply.run_command
+    end
+
+    describe "when a filesystem lock is encountered" do
+      before :each do
+        @apply.expects(:lock).raises(Puppet::LockError, 'locked')
+        Puppet.expects(:notice).with(regexp_matches /Run of Puppet already in progress; skipping  (.+ exists)/)
+      end
+
+      it "should exit" do
+        expect { @apply.run_command }.to raise_error(SystemExit)
```

Comment:
Found out from @joshcooper that there's a helper for this, https://github.com/puppetlabs/puppet/blob/master/spec/integration/application/apply_spec.rb#L38, so we can drop lines 192-201 down to one spec test.